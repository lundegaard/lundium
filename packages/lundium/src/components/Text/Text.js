import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';

import classNamesByBreakpoint from '../../utils/classNamesByBreakpoint';
import Box from '../Box';

const alignToClassName = classNamesByBreakpoint((align, breakpoint) =>
	breakpoint === 'xs' ? `text-${align}` : `text-${breakpoint}-${align}`,
);

const Text = forwardRef(
	(
		{
			align,
			as = 'p',
			children,
			wrap,
			transform,
			className,
			color,
			weight,
			size,
			...rest
		},
		ref,
	) => (
		<Box
			ref={ref}
			as={as}
			{...rest}
			className={cx(
				className,
				{
					[`text-${wrap}`]: wrap,
					[`text-${transform}`]: transform,
					[`text-${color}`]: color,
					[`text-${size}`]: size,
					[`font-weight-${weight}`]: weight,
				},
				alignToClassName && alignToClassName(align),
			)}
		>
			{children}
		</Box>
	),
);
Text.displayName = 'forwardRef(Text)';

Text.propTypes = {
	align: PropTypes.oneOfType([
		PropTypes.object,
		PropTypes.arrayOf(
			PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
		),
		PropTypes.oneOf(['left', 'center', 'right']),
	]),
	/** Tag used for root component */
	as: PropTypes.any,
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** Class name for root. */
	className: PropTypes.string,
	color: PropTypes.string,
	size: PropTypes.oneOf(['md', 'lg']),
	transform: PropTypes.oneOf(['lowercase', 'uppercase', 'capitalize']),
	weight: PropTypes.oneOf([
		'extra-bold',
		'bold',
		'semibold',
		'normal',
		'light',
	]),
	wrap: PropTypes.oneOf(['wrap', 'nowrap', 'truncate']),
	...Box.propTypes,
};

export default Text;
