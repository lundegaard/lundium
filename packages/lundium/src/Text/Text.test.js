import React from 'react';
import { shallow } from 'enzyme';

import Text from './Text';

describe('<Text />', () => {
	const requiredProps = {};

	it('shallow', () => {
		const wrapper = shallow(<Text {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
