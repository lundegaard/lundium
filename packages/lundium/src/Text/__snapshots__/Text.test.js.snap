// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> shallow 1`] = `
<forwardRef(Box)
  as="p"
  className=""
/>
`;
